components/style/sharing/mod.rs

```diff
@@ -228,11 +431,6 @@ impl<E: TElement> StyleSharingCandidateCache<E> {
             return StyleSharingResult::CannotShare
         }
 
-        if element.get_id().is_some() {
```

Comment:
This should've landed when Boris landed the `element.get_id() != candidate.get_id()` changes not long ago.